segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/WriteOperation.java

```diff
@@ -188,30 +187,28 @@ private void collectGarbage() {
 
         // if layout did not change then commit with lazyWrite.
         return txn.commit(!didSegmentLayoutChange && chunkedSegmentStorage.getConfig().isLazyCommitEnabled())
-                .thenRunAsync(() -> {
-                    isCommitted = true;
-                }, chunkedSegmentStorage.getExecutor());
+                .thenRunAsync(() -> isCommitted = true, chunkedSegmentStorage.getExecutor());
```

Comment:
Do you really need to run this on another thread? `thenRun` will suffice. 

Everywhere else too.

Reply:
thenRun will potentially run on callers thread which may or may not include segment store core threads.
I do want Storage code to **always** run on storage pool.


With thread pools the number of threads are fixed and we are not switching threads explicitly here. Framework is free to schedule any task on any free thread- including then thenRun. It uses variant of work stealing which will greedily execute next available task.

Reply:
There is no extra overhead here.